lib/tenderjit/runtime.rb

```diff
@@ -510,9 +517,11 @@ def RB_BUILTIN_TYPE loc
     def temp_var name = "temp_var"
       tv = TemporaryVariable.new @fisk.register(name), Fiddle::TYPE_VOIDP, Fiddle::SIZEOF_VOIDP, 0, self
 
+      @active_temp_vars << tv
+
       if block_given?
         yield tv
-        tv.release!
+        self.release_temp(tv)
       else
         tv
```

Comment:
I haven't tested this, but maybe someone is calling `tmp_var` without a block and then not releasing it "when we want"?  Can you change this to always yield and then find who is using it without a block?  (maybe we should require a block always?)

Reply:
Thanks for the suggestion! I'll study more in depth the allocation; I have the impression I'm missing something subtle. I have some ideas, but if start trying it now it's going to be a night of hair loss :joy: 

> maybe we should require a block always?

If we exclude the case where a dev forgets to release a variable (it can be excluded because it causes an error), I guess that the user decision is not really affected by the fact the release is via command or via end of block; but that's how I think, I don't know if that can be generalized :smile: From an engineering perspective, command or end-of-block release both end invoking the same method.

Allowing only blocks though, would have regardless the merit of simplifying temp vars usage to a single form.

Something to keep in mind is the edge case of two branches where a temp var is released with different times. In this case, using a block won't do. But this is theoretical; I'll need to verify the existing code.

Reply:
> Can you change this to always yield and then find who is using it without a block? (maybe we should require a block always?)

In relation to this, I've actually found the edge case I was suspecting, where [two branches release a temp var differently](https://github.com/64kramsystem/tenderjit-dev/blob/main/lib/tenderjit/iseq_compiler.rb#L735-L761). I was actually wondering if this is unintended (and masked by the exit), because the else branch doesn't release the variable.

Reply:
Both sides get evaluated at compile time because it's generating machine code for the branch, so it shouldn't matter.  It is very confusing to read though, so it's probably better if we make this use the block form or pull the `release!` line below the else case.

Reply:
I'm going to handle this in a separate PR. Knowing that pushing the release at the end won't hurt, we may have the option to entirely remove the manual temp var release.